vnext/Shared/InspectorPackagerConnection.cpp

```diff
@@ -0,0 +1,216 @@
+
+#include "pch.h"
+
+#include <folly/json.h>
+#include "InspectorPackagerConnection.h"
+#include <tracing/tracing.h>
+
+namespace Microsoft::ReactNative {
+
+namespace {
+
+struct InspectorProtocol {
+  static constexpr const char InspectorMessage_EVENT[] = "event";
+  static constexpr const char InspectorMessage_PAGEID[] = "pageId";
+  static constexpr const char InspectorMessage_PAYLOAD[] = "payload";
+
+  static constexpr const char InspectorMessage_eventName_wrappedEvent[] = "wrappedEvent";
+  static constexpr const char InspectorMessage_eventName_getPages[] = "getPages";
+  static constexpr const char InspectorMessage_eventName_connect[] = "connect";
+  static constexpr const char InspectorMessage_eventName_disconnect[] = "disconnect";
+
+  enum class EventType { GetPages, WrappedEvent, Connect, Disconnect };
+
+  static EventType getEventType(const folly::dynamic &messageFromPackager) {
+    std::string event = messageFromPackager.at(InspectorProtocol::InspectorMessage_EVENT).getString();
+    if (0 == event.compare(InspectorMessage_eventName_getPages))
+      return EventType::GetPages;
+    if (0 == event.compare(InspectorMessage_eventName_wrappedEvent))
+      return EventType::WrappedEvent;
+    if (0 == event.compare(InspectorMessage_eventName_connect))
+      return EventType::Connect;
+    if (0 == event.compare(InspectorMessage_eventName_disconnect))
+      return EventType::Disconnect;
+    assert(false && "Unknown event!");
+    std::abort();
+  }
+
+  static folly::dynamic constructResponseForPackager(EventType eventType, folly::dynamic &&payload) {
+    folly::dynamic response = folly::dynamic::object;
```

Comment:
isn't this supposed to be `folly::dynamic::object()` ?

Reply:
figured it out; `dynamic` has a ctor that takes a "function that returns ObjectMaker", which `folly::dynamic::object` is. Unintuitive but obviously it works :)